### PR TITLE
Add support for status checks using `vc-status-list@1.0`

### PR DESCRIPTION
**Pull request by JSAssassin**

### CODE REVIEW

**Review by msporny — APPROVED** [review]

File: lib/status.js [hunk 1]
```diff
@@ -22,9 +35,12 @@ exports.checkStatus = async function(options = {}) {
       // no status to check
       return {verified: true};
     }
-
-    if(statusTypeMatches({credential})) {
-      result = await checkStatus(options);
+    const credentialStatusTypeMatches =
+      statusTypeMatches.get(credentialStatus.type);
+    if(credentialStatusTypeMatches &&
+        credentialStatusTypeMatches({credential})) {
```
Comment (dlongley) [inline lib/status.js:41]:
```suggestion
      credentialStatusTypeMatches({credential})) {
```

File: test/mocha/40-credential-status.js [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+revokedRlCredential.credentialSubject.encodedList =
+  encodedList100KWith50KthRevoked;
+
+const revokedUnsignedCredential2 = clone(unsignedCredentialRL2020Type);
+revokedUnsignedCredential2.credentialStatus.id =
+  `${revokedRlCredential.id}/50000`;
+revokedUnsignedCredential2.credentialStatus.revocationListIndex = 50000;
+revokedUnsignedCredential2.credentialStatus.revocationListCredential =
+  `${revokedRlCredential.id}/50000`;
```
Comment (dlongley) [inline test/mocha/40-credential-status.js:154]: The status and the list credential can't have the same ID -- they aren't the same thing.

File: test/mocha/40-credential-status.js [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+revokedUnsignedCredential2.issuer = revokedRlCredential.issuer;
+
+// mount the test routes
+app.get('/status/1',
+  // eslint-disable-next-line no-unused-vars
+  (req, res, next) => {
+    // responds with a valid status list 2021 type credential
+    res.json(slCredential);
+  });
+app.get('/status/1/50000',
```
Comment (dlongley) [inline test/mocha/40-credential-status.js:164]: This approach seems to defeat the privacy purpose of the list design and may be flagged as such -- or erroneously copied by someone else. We may even want to reject status IDs that do this sort of thing and require them to use hash fragments instead.

**Review by dlongley — CHANGES_REQUESTED** [review]

**Review by dlongley — APPROVED** [review]
RSLGTM.

**Pull request merged**